enumerable_blocks.rb

```diff
@@ -0,0 +1,157 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  # my_each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    element = if instance_of?(Array)
+                self
+              else
+                to_a
+              end
+
+    k = 0
+    while k < element.length
+      yield element[k]
+      k += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    element = to_a
+    k = 0
+    while k < element.length
+      yield element[k], k
+      k += 1
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield i }
+    arr
+  end
+
+  # my_all?
+  def my_all?(argument = nil)
+    if block_given? && !argument.nil?
+      return false unless my_all?(argument) && my_all?
+    end
+    if block_given?
+      my_each { |item| return false unless yield(item) }
+    elsif argument.nil?
+      my_each { |item| return false unless item }
+    elsif argument.is_a?(Class)
+      my_each { |item| return false unless item.class.is_a?(Class) }
+    elsif argument.class == Regexp
+      my_each { |item| return false unless item.match(argument) }
+    else
+      my_each { |item| return false unless item == argument && item.class <= argument.class }
+    end
+    true
+  end
```

Comment:
- [ ] `my_all` is not working as the original `all?` method, also is not passing all test cases. 

### `all?` is an Enumerable that has these characteristics:
- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending if the values in the array/range/hash match the condition

```ruby
[1,2,false].all?(Numeric)  # false
[1,2,false].my_all?(Numeric) # true

[1,2,3].all?(String) # false
[1,2,3].my_all?(String) # true
```

[Enumerable#all?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-all-3F)